### ISSUE TITLE
Using openai's opensrc model

### ISSUE THREAD

**Issue by Abc11c** [NVIDIA/garak#1327]
I'm trying to test out the new openai's open src model 

```
 python -m garak   --model_type huggingface   --model_name openai/gpt-oss-20b   --probes divergence   --generator_options '{"hf_args
":{"max_length":4096,"trust_remote_code":true,"torch_dtype":"bfloat16"}}'
garak LLM vulnerability scanner v0.12.0.pre1 ( https://github.com/NVIDIA/garak ) at 2025-08-08T23:41:25.240335
📜 logging to /home/xxx/.local/share/garak/garak.log
🦜 loading generator: Hugging Face 🤗 pipeline: openai/gpt-oss-20b
MXFP4 quantization requires triton >= 3.4.0 and triton_kernels installed, we will default to dequantizing the model to bf16
Loading checkpoint shards: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 3/3 [00:07<00:00,  2.49s/it]
Device set to use cuda
📜 reporting to /home/xxx/.local/share/garak/garak_runs/garak.eabe3c03-1f09-4c10-9c49-ba6ed0361a58.report.jsonl
🕵️  queue of probes: divergence.Repeat
probes.divergence.Repeat:   0%|                                                                                                                           | 0/36 [00:00<?, ?it/s]Asking to truncate to max_length but no maximum length is provided and the model has no predefined maximum length. Default to no truncation.
probes.divergence.Repeat:   3%|███▏                                                                                                               | 1/36 [00:00<00:09,  3.70it/s]You seem to be using the pipelines sequentially on GPU. In order to maximize efficiency please use a dataset
divergence.Repeat                                                          divergence.RepeatDiverges: SKIP  ok on    0/   0                                                      
divergence.Repeat                                                        mitigation.MitigationBypass: SKIP  ok on    0/   0
📜 report closed :) /home/xxx/.local/share/garak/garak_runs/garak.eabe3c03-1f09-4c10-9c49-ba6ed0361a58.report.jsonl
📜 report html summary being written to /home/xxx/.local/share/garak/garak_runs/garak.eabe3c03-1f09-4c10-9c49-ba6ed0361a58.report.html
✔️  garak run complete in 12.35s
```

This is the log file, how do I debug or any advice on how to proceed  ? Thanks! 
```
2025-08-08 23:45:18,541  INFO  invoked
2025-08-08 23:45:18,542  DEBUG  Loading configs from: /home/xxx/working/redteam/garak/garak/resources/garak.core.yaml
2025-08-08 23:45:18,546  DEBUG  args - raw argument string received: ['--model_type', 'huggingface', '--model_name', 'openai/gpt-oss-20b', '--probes', 'divergence', '--generator_options', '{"hf_args":{"max_length":4096,"trust_remote_code":true,"torch_dtype":"bfloat16"}}']
2025-08-08 23:45:18,546  DEBUG  args - full argparse: Namespace(verbose=0, report_prefix=None, narrow_output=False, parallel_requests=False, parallel_attempts=False, skip_unknown=False, seed=None, deprefix=True, eval_threshold=0.5, generations=5, config=None, model_type='huggingface', model_name='openai/gpt-oss-20b', probes='divergence', probe_tags=None, detectors='auto', extended_detectors=False, buffs=None, buff_option_file=None, buff_options=None, detector_option_file=None, detector_options=None, generator_option_file=None, generator_options='{"hf_args":{"max_length":4096,"trust_remote_code":true,"torch_dtype":"bfloat16"}}', harness_option_file=None, harness_options=None, probe_option_file=None, probe_options=None, taxonomy=None, plugin_info=None, list_probes=False, list_detectors=False, list_generators=False, list_buffs=False, list_config=False, version=False, report=None, interactive=False, generate_autodan=False, fix=False, **{'interactive.py': False})
2025-08-08 23:45:18,676  DEBUG  no site config found at: /home/xxx/.config/garak/garak.site.yaml
2025-08-08 23:45:18,676  DEBUG  Loading configs from: /home/xxx/working/redteam/garak/garak/resources/garak.core.yaml
2025-08-08 23:45:18,679  DEBUG  args - cli_args&commands stored: Namespace(model_type='huggingface', model_name='openai/gpt-oss-20b', probes='divergence', generator_options='{"hf_args":{"max_length":4096,"trust_remote_code":true,"torch_dtype":"bfloat16"}}', verbose=0, list_detectors=False, list_probes=False, list_generators=False, list_buffs=False, list_config=False, plugin_info=None, interactive=False, report=None, version=False, fix=False)
2025-08-08 23:45:18,679  DEBUG  non-config params: [('probes', 'divergence'), ('generator_options', '{"hf_args":{"max_length":4096,"trust_remote_code":true,"torch_dtype":"bfloat16"}}'), ('list_detectors', False), ('list_probes', False), ('list_generators', False), ('list_buffs', False), ('list_config', False), ('plugin_info', None), ('report', None), ('version', False), ('fix', False)]
2025-08-08 23:45:19,534  INFO  generator init: <garak.generators.huggingface.Pipeline object at 0x76b52931c190>
2025-08-08 23:45:19,573  DEBUG  Using cuda, based on torch environment evaluation
2025-08-08 23:45:20,706  DEBUG  Starting new HTTPS connection (1): huggingface.co:443
2025-08-08 23:45:20,786  DEBUG  https://huggingface.co:443 "HEAD /openai/gpt-oss-20b/resolve/main/config.json HTTP/1.1" 307 0
2025-08-08 23:45:20,794  DEBUG  https://huggingface.co:443 "HEAD /api/resolve-cache/models/openai/gpt-oss-20b/cbf31f62664d4b1360b3a78427f7b3c3ed8f0fa8/config.json HTTP/1.1" 200 0
2025-08-08 23:45:24,826  DEBUG  https://huggingface.co:443 "HEAD /openai/gpt-oss-20b/resolve/main/generation_config.json HTTP/1.1" 307 0
2025-08-08 23:45:24,835  DEBUG  https://huggingface.co:443 "HEAD /api/resolve-cache/models/openai/gpt-oss-20b/cbf31f62664d4b1360b3a78427f7b3c3ed8f0fa8/generation_config.json HTTP/1.1" 200 0
2025-08-08 23:45:24,871  DEBUG  https://huggingface.co:443 "HEAD /openai/gpt-oss-20b/resolve/main/custom_generate/generate.py HTTP/1.1" 404 0
2025-08-08 23:45:25,099  DEBUG  https://huggingface.co:443 "GET /api/models/openai/gpt-oss-20b/tree/main/additional_chat_templates?recursive=False&expand=False HTTP/1.1" 404 64
2025-08-08 23:45:25,622  INFO  run started at 2025-08-08T23:45:18.539374
2025-08-08 23:45:25,623  DEBUG  relative report dir provided
2025-08-08 23:45:25,623  INFO  reporting to /home/xxx/.local/share/garak/garak_runs/garak.9b2bfc66-3211-4969-afb6-27ea934cf934.report.jsonl
2025-08-08 23:45:25,624  INFO  service import: garak.langservice
2025-08-08 23:45:25,772  INFO  harness init: <garak.harnesses.probewise.ProbewiseHarness object at 0x76b3e4c2ea40>
2025-08-08 23:45:25,772  INFO  probe queue: probes.divergence.Repeat
2025-08-08 23:45:25,773  INFO  probe init: <garak.probes.divergence.Repeat object at 0x76b3cd8cd420>
2025-08-08 23:45:25,773  DEBUG  langauge provision service: en,en
2025-08-08 23:45:25,773  INFO  detector init: <garak.detectors.divergence.RepeatDiverges object at 0x76b3cd8cd930>
2025-08-08 23:45:25,773  INFO  detector init: <garak.detectors.mitigation.MitigationBypass object at 0x76b3bd2e44c0>
2025-08-08 23:45:25,773  INFO  detector init: <garak.detectors.mitigation.MitigationBypass object at 0x76b3bd2e4940>
2025-08-08 23:45:25,773  DEBUG  harness: probe start for garak.probes.divergence.Repeat
2025-08-08 23:45:25,774  DEBUG  probe execute: <garak.probes.divergence.Repeat object at 0x76b3cd8cd420>
2025-08-08 23:45:25,992  ERROR  expected scalar type Half but found BFloat16
2025-08-08 23:45:26,000  ERROR  expected scalar type Half but found BFloat16
2025-08-08 23:45:26,008  ERROR  expected scalar type Half but found BFloat16
2025-08-08 23:45:26,015  ERROR  expected scalar type Half but found BFloat16
2025-08-08 23:45:26,022  ERROR  expected scalar type Half but found BFloat16
2025-08-08 23:45:26,029  ERROR  expected scalar type Half but found BFloat16
2025-08-08 23:45:26,037  ERROR  expected scalar type Half but found BFloat16
2025-08-08 23:45:26,044  ERROR  expected scalar type Half but found BFloat16
2025-08-08 23:45:26,051  ERROR  expected scalar type Half but found BFloat16
2025-08-08 23:45:26,058  ERROR  expected scalar type Half but found BFloat16
2025-08-08 23:45:26,065  ERROR  expected scalar type Half but found BFloat16
2025-08-08 23:45:26,072  ERROR  expected scalar type Half but found BFloat16
2025-08-08 23:45:26,079  ERROR  expected scalar type Half but found BFloat16
2025-08-08 23:45:26,086  ERROR  expected scalar type Half but found BFloat16
2025-08-08 23:45:26,093  ERROR  expected scalar type Half but found BFloat16
2025-08-08 23:45:26,100  ERROR  expected scalar type Half but found BFloat16
2025-08-08 23:45:26,107  ERROR  expected scalar type Half but found BFloat16
2025-08-08 23:45:26,114  ERROR  expected scalar type Half but found BFloat16
2025-08-08 23:45:26,121  ERROR  expected scalar type Half but found BFloat16
2025-08-08 23:45:26,129  ERROR  expected scalar type Half but found BFloat16
2025-08-08 23:45:26,136  ERROR  expected scalar type Half but found BFloat16
2025-08-08 23:45:26,143  ERROR  expected scalar type Half but found BFloat16
2025-08-08 23:45:26,150  ERROR  expected scalar type Half but found BFloat16
2025-08-08 23:45:26,157  ERROR  expected scalar type Half but found BFloat16
2025-08-08 23:45:26,164  ERROR  expected scalar type Half but found BFloat16
2025-08-08 23:45:26,171  ERROR  expected scalar type Half but found BFloat16
2025-08-08 23:45:26,178  ERROR  expected scalar type Half but found BFloat16
2025-08-08 23:45:26,185  ERROR  expected scalar type Half but found BFloat16
2025-08-08 23:45:26,192  ERROR  expected scalar type Half but found BFloat16
2025-08-08 23:45:26,199  ERROR  expected scalar type Half but found BFloat16
2025-08-08 23:45:26,207  ERROR  expected scalar type Half but found BFloat16
2025-08-08 23:45:26,214  ERROR  expected scalar type Half but found BFloat16
2025-08-08 23:45:26,221  ERROR  expected scalar type Half but found BFloat16
2025-08-08 23:45:26,228  ERROR  expected scalar type Half but found BFloat16
2025-08-08 23:45:26,235  ERROR  expected scalar type Half but found BFloat16
2025-08-08 23:45:26,241  ERROR  expected scalar type Half but found BFloat16
2025-08-08 23:45:26,242  DEBUG  probe return: <garak.probes.divergence.Repeat object at 0x76b3cd8cd420> with 36 attempts
2025-08-08 23:45:26,242  DEBUG  harness: run detector garak.detectors.divergence.RepeatDiverges
2025-08-08 23:45:26,242  DEBUG  harness: run detector garak.detectors.mitigation.MitigationBypass
2025-08-08 23:45:26,243  DEBUG  harness: probe list iteration completed
2025-08-08 23:45:26,243  INFO  run complete, ending
2025-08-08 23:45:26,260  INFO  garak run complete in 7.70s
```

**Comment by qrdlgit** [comment]
I was able to get it to work, though I forked the repo and made some changes: https://www.kaggle.com/code/kaggleqrdl/router-logits-visualization/notebook   I attached the logs as a dataset.   I'm looking into GCG and TAP next.

I can't say for sure what your issue is, but I think using huggingface.Model might be a good start.

**Comment by jmartin-tech** [comment]
The stored default config shipped with `openai/gpt-oss-20b` is not friendly to use with the default `huggingface.Pipeline` generator at this time. I suspect the stored config is too specific to the hardware used during training. Please use `huggingface.Model` for early adoption testing.

Also the generator configuration options passed needs to specify the generator type in the json structure. The following works for a system with a cuda capable GPU installed and a fresh conda env of `0.12.0`:
```
python -m garak --model_type huggingface.Model --model_name openai/gpt-oss-20b --probes divergence --generator_options '{"huggingface":{"Model":{"hf_args":{"device_map":"cuda"}}}}'
```

**Comment by jmartin-tech** [comment]
Closing as answered.
